llvm/lib/IR/RepoTicket.cpp

```diff
@@ -116,78 +125,135 @@ static GODigestState accumulateGOFinalDigestOrInitialDigestAndGetGODigestState(
   return accumulateGOInitialDigestAndGetGODigestState(GO, GOHash, GOIMap);
 }
 
-// Update the GO's hash value by adding the hash of its dependents.
-template <typename Function>
-static bool updateDigestUseDependenciesAndContributions(
-    const GlobalObject *GO, MD5 &GOHash, GOStateMap &Visited, GOInfoMap &GOIMap,
-    Function AccumulateGODigest) {
+// Update the digest of an invidual GO incorporating the hashes of all its
+// dependencies and contributions.
+// \param GO The global object whose digest is to be computed.
+// \param Visited A map of GO to a unique number in the dependencies or
+//        contributions graph.
+// \param GOIMap A map of GO to GOInfo
+// \param GOHashCache Used to record memoized hashes.
+// \param AccumulateGODigest A function is used to accumulate the GO's digest.
+// \returns a tuple containing the state information which includes 1) return
+// true if the module M has been changed, 2) a numerical identifier which will
+// be used if GO loops back to itself in future and 3) the GO's digest.
+template <typename FunctionType>
+static auto updateDigestUseDependenciesAndContributions(
+    const GlobalObject *GO, GOStateMap &Visited, GOInfoMap &GOIMap,
+    MemoizedHashes &GOHashCache, FunctionType AccumulateGODigest)
+    -> std::tuple<bool, size_t, DigestType> {
   if (GO->isDeclaration())
-    return false;
+    return std::make_tuple(false, std::numeric_limits<size_t>::max(),
+                           NullDigest);
+
+  const auto NumVisited = Visited.size();
```

Comment:
NumVisited is largely used as an index/identifier rather than a count so maybe it should be named to reflect that, maybe GOIndex or GOVisitedIndex.

Reply:
Yes, changed to GOVisitedIndex.